packages/core/src/app/payment/paymentMethod/BraintreeAchPaymentForm.tsx

```diff
@@ -0,0 +1,350 @@
+        custom: false,
+        id: BraintreeAchBankAccountValues.OwnershipType,
+        label: 'Ownership Type',
+        required: true,
+        fieldType: DynamicFormFieldType.dropdown,
+        options: {
+            items: ownershipTypeOptions
+        }
+    },
+    {
+        name: BraintreeAchBankAccountValues.FirstName,
+        custom: false,
+        id: BraintreeAchBankAccountValues.FirstName,
+        label: 'First Name',
+        required: true,
+    },
+    {
+        name: BraintreeAchBankAccountValues.LastName,
+        custom: false,
+        id: BraintreeAchBankAccountValues.LastName,
+        label: 'Last Name',
+        required: true,
+    },
+    {
+        name: BraintreeAchBankAccountValues.BusinessName,
+        custom: false,
+        id: BraintreeAchBankAccountValues.BusinessName,
+        label: 'Business Name',
+        required: true,
+    },
+    {
+        name: BraintreeAchBankAccountValues.Address1,
+        custom: false,
+        id: BraintreeAchBankAccountValues.Address1,
+        label: 'Address',
+        required: true,
+    },
+    {
+        name: BraintreeAchBankAccountValues.Address2,
+        custom: false,
+        id: BraintreeAchBankAccountValues.Address2,
+        label: 'Apartment/Suite/Building',
+        required: false,
+    },
+    {
+        name: BraintreeAchBankAccountValues.StateOrProvinceCode,
+        custom: false,
+        id: BraintreeAchBankAccountValues.StateOrProvinceCode,
+        label: 'State/Province',
+        required: true,
+        fieldType: DynamicFormFieldType.dropdown,
+    },
+    {
+        name: BraintreeAchBankAccountValues.City,
+        custom: false,
+        id: BraintreeAchBankAccountValues.City,
+        label: 'City',
+        required: true,
+    },
+    {
+        name: BraintreeAchBankAccountValues.PostalCode,
+        custom: false,
+        id: BraintreeAchBankAccountValues.PostalCode,
+        label: 'Postal Code',
+        required: true,
+    },
+];
+
+export interface AchFormFieldsProps {
+    ownershipType: string;
+    handleChange: (fieldId: string) => (value: string) => void;
+    provinceOptions?: Array<{ value: string; label: string }>
+}
+
+const AchFormFields: FunctionComponent<AchFormFieldsProps> = ({
+    ownershipType,
+    handleChange,
+    provinceOptions,
+}) => {
+    const isBusiness = useMemo(() => ownershipType === OwnershipTypes.Business, [ownershipType]);
+
+    return (
+        <>
+            {useMemo(() => (
+                formFieldData.filter(({ id }) => (
+                    isBusiness ? id !== BraintreeAchBankAccountValues.FirstName && id !== BraintreeAchBankAccountValues.LastName : id !== BraintreeAchBankAccountValues.BusinessName
+                )).map((field) => (
+                    <DynamicFormField
+                        extraClass={`dynamic-form-field--${field.id ?? field.name}`}
+                        field={field.name === BraintreeAchBankAccountValues.StateOrProvinceCode ? {
```

Comment:
Would be better remove this complexity from the render. You can compose filtered and ready to consumption formFieldData inside [mapFromCheckoutProps](https://github.com/bigcommerce/checkout-js/pull/1221/files#diff-54d895cbdbf35a5073f2e2083d5da2c1ee6169b119fa629bd3e28b1d37d6cbcdR319) and just use here formFieldData prop.

Reply:
I created HOC **withAchBraintreeFields** and move all these logic there